erpnext/hr/doctype/leave_application/leave_application.py

```diff
@@ -152,8 +158,22 @@ def update_attendance(self):
 					doc.leave_application = self.name
 					doc.status = status
 					doc.flags.ignore_validate = True
+					doc.remaining_half_day_status = self.remaining_half_day_status if status == half_day_status else ''
 					doc.insert(ignore_permissions=True)
 					doc.submit()
+		else:
+			frappe.throw(_("Please Select Applicable status for Leave application in Attendance Status"))
+
+	def get_applicable_status_for_Attendance(self):
+		statuses = frappe.get_all("Attendance Status", filters = {"applicable_for_leave_application"}, fields=["name","is_half_day", "is_leave"])
+
+		half_day_status = [status.name for status in statuses if status.is_half_day == 1]
+		on_leave_status = [status.name for status in statuses if status.is_leave == 1]
```

Comment:
```suggestion
		half_day_status = [status.name for status in statuses if status.is_half_day]
		on_leave_status = [status.name for status in statuses if status.is_leave]
```
The if statement will return 1 only so no need to check explicitly.